src/compressor/Compressor.h

```diff
@@ -44,7 +43,8 @@ class Compressor {
     COMP_ALG_LAST   //the last value for range checks
   };
 
-  inline static const std::map<const std::string, const CompressionAlgorithm> compression_algorithms {
+  using pair_type = std::pair<const char*, CompressionAlgorithm>;
+  static constexpr std::initializer_list<pair_type> compression_algorithms {
```

Comment:
nit, the name of "compression_algorithms" is a little bit redundant. as the class name of this method is `Compressor`, probably we could just name it `algorithms`.